UPGRADING.md

```diff
@@ -1,6 +1,51 @@
 Upgrading
 =========
 
+## 7.x to 8.x
+
+### node
+
+This version contains major improvements to the node notifier, including making the top-level `Bugsnag` static interface context-aware and enabling breadcrumbs for node projects.
+
+The minimum supported node version is now v12.17.0.
+
+#### context-aware `Bugsnag` calls
+
+When using `plugin-express`, `plugin-koa`, `plugin-restify`, or `plugin-contextualize`, a clone of the top-level Bugsnag client is made so that any subsequent changes made to the client (such as attaching metadata) only affect the scope of the current web request (or a function call, when using `plugin-contextualize`).
+
+Prior to `bugsnag-js` v8, calls made to the top-level `Bugsnag` static interface were not aware of this context so users had to ensure they were calling methods on the correct client instance, i.e. the cloned client that was made available on `req.bugsnag` (or `ctx.bugsnag` for koa). This wasn't ideal because if you wanted to interact with Bugsnag in some function deep in a call stack you would have to pass `req.bugsnag` all the way down, as calling `Bugsnag.notify` would not have contained the request metadata gathered by the plugin. With version 8 of the notifier, top-level calls to `Bugsnag` are now context-aware. This means you can call `Bugsnag.notify` (or `Bugsnag.leaveBreadcrumb` etc.), and, if it was called within a context, the call will be forwarded to the correct cloned version of that client (i.e. for the particular request from which the call originated).
+
+Express
+
+```diff
+app.get('/handled', function (req, res) {
+- req.bugsnag.notify(new Error('handled'))
++ Bugsnag.notify(new Error('handled'))
+})
+```
+
+Koa
+
+```diff
+app.use(async (ctx, next) => {
+  if (ctx.path === '/handled') {
+-    ctx.bugsnag.notify(new Error('handled'))
++    Bugsnag.notify(new Error('handled'))
+    await next()
+  } else {
+    await next()
+  }
+})
+```
+
+Note: `req.bugsnag` (and `ctx.bugsnag` in koa) is still present in version 8 of `bugsnag-js`.
```

Comment:
I wonder if we want to say something:
- "you can continue using req.bugsnag as before"
- there are situations where context is lost and you may need to use `req.bugsnag`. See the context-loss section of the docs

Reply:
```suggestion
Notes

* `req.bugsnag` (and `ctx.bugsnag` in koa) is still present in version 8 of `bugsnag-js`, so you can continue using these as before.
* There are rare situations on Express servers when this contextual storage can get lost, causing the data stored to become server-scoped and so affect all threads that are being executed. See our [online docs](https://docs.bugsnag.com/platforms/javascript/express/node-async/#context-loss-in-express-servers) for full details.
```